modules/lora_basics_modem/common.cmake

```diff
@@ -0,0 +1,394 @@
+# Common
+#-----------------------------------------------------------------------------
+
+zephyr_compile_definitions(NUMBER_OF_STACKS=${CONFIG_LORA_BASICS_MODEM_NUMBER_OF_STACKS})
+
+# SMTC_MODEM_CORE_C_SOURCES
+zephyr_library_sources(
+  ${LBM_SMTC_MODEM_CORE_DIR}/lorawan_api/lorawan_api.c
+  ${LBM_SMTC_MODEM_CORE_DIR}/smtc_modem.c
+  ${LBM_SMTC_MODEM_CORE_DIR}/smtc_modem_test.c
+  ${LBM_SMTC_MODEM_CORE_DIR}/modem_utilities/modem_event_utilities.c
+  ${LBM_SMTC_MODEM_CORE_DIR}/modem_utilities/fifo_ctrl.c
+  ${LBM_SMTC_MODEM_CORE_DIR}/modem_utilities/modem_core.c
+  ${LBM_SMTC_MODEM_CORE_DIR}/modem_supervisor/modem_supervisor_light.c
+  ${LBM_SMTC_MODEM_CORE_DIR}/modem_supervisor/modem_tx_protocol_manager.c
+  ${LBM_SMTC_MODEM_CORE_DIR}/lorawan_packages/lorawan_certification/lorawan_certification.c
+  ${LBM_SMTC_MODEM_CORE_DIR}/lorawan_manager/lorawan_join_management.c
+  ${LBM_SMTC_MODEM_CORE_DIR}/lorawan_manager/lorawan_send_management.c
+  ${LBM_SMTC_MODEM_CORE_DIR}/lorawan_manager/lorawan_cid_request_management.c
+  ${LBM_SMTC_MODEM_CORE_DIR}/lorawan_manager/lorawan_dwn_ack_management.c
+)
+
+# LR1MAC_C_SOURCES
+zephyr_library_sources(
+  ${LBM_SMTC_MODEM_CORE_DIR}/lr1mac/src/lr1_stack_mac_layer.c
+  ${LBM_SMTC_MODEM_CORE_DIR}/lr1mac/src/lr1mac_core.c
+  ${LBM_SMTC_MODEM_CORE_DIR}/lr1mac/src/lr1mac_utilities.c
+  ${LBM_SMTC_MODEM_CORE_DIR}/lr1mac/src/smtc_real/src/smtc_real.c
+  ${LBM_SMTC_MODEM_CORE_DIR}/lr1mac/src/services/smtc_duty_cycle.c
+  ${LBM_SMTC_MODEM_CORE_DIR}/lr1mac/src/services/smtc_lbt.c
+)
+
+# SMTC_MODEM_CRYPTO_C_SOURCES
+zephyr_library_sources(
+  ${LBM_SMTC_MODEM_CORE_DIR}/smtc_modem_crypto/smtc_modem_crypto.c
+)
+
+# RADIO_PLANNER_C_SOURCES
+zephyr_library_sources(
+  ${LBM_SMTC_MODEM_CORE_DIR}/radio_planner/src/radio_planner.c
+)
+
+zephyr_include_directories(
+  ${LORA_BASICS_MODEM_LIB_DIR}
+  ${LORA_BASICS_MODEM_LIB_DIR}/smtc_modem_api
+  ${LORA_BASICS_MODEM_LIB_DIR}/smtc_modem_hal
+  ${LBM_SMTC_MODEM_CORE_DIR}
+  ${LBM_SMTC_MODEM_CORE_DIR}/lorawan_api
+  ${LBM_SMTC_MODEM_CORE_DIR}/lorawan_manager
+  ${LBM_SMTC_MODEM_CORE_DIR}/lorawan_packages
+  ${LBM_SMTC_MODEM_CORE_DIR}/lorawan_packages/lorawan_certification
+  ${LBM_SMTC_MODEM_CORE_DIR}/lr1mac
+  ${LBM_SMTC_MODEM_CORE_DIR}/lr1mac/src
+  ${LBM_SMTC_MODEM_CORE_DIR}/lr1mac/src/services
+  ${LBM_SMTC_MODEM_CORE_DIR}/lr1mac/src/smtc_real/src
+  ${LBM_SMTC_MODEM_CORE_DIR}/modem_supervisor
+  ${LBM_SMTC_MODEM_CORE_DIR}/modem_utilities
+  ${LBM_SMTC_MODEM_CORE_DIR}/radio_planner/src
+  ${LBM_SMTC_MODEM_CORE_DIR}/smtc_modem_crypto
+  ${LBM_SMTC_MODEM_CORE_DIR}/smtc_modem_crypto/smtc_secure_element
+  ${LBM_SMTC_MODEM_CORE_DIR}/smtc_ral/src
+  ${LBM_SMTC_MODEM_CORE_DIR}/smtc_ralf/src
+)
+
+#-----------------------------------------------------------------------------
+# ALCSync
+#-----------------------------------------------------------------------------
+
+zephyr_compile_definitions_ifdef(CONFIG_LORA_BASICS_MODEM_ALC_SYNC
```

Comment:
Yes, that is actually something I wanted to check (there are some `ifdef`s in the public header, so that's something to be done with caution).
Thank you for the reminder, I had forgot about that.

Reply:
See 29821ec2d1758c445c3b0cba9a914db852fc95cc and 2fd9f86d997abf38f31989cdcea8c3e74fb346b9.

(i'll squash the commits when all issues are fixed)